app/main.py

```diff
@@ -1,8 +1,24 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_instances = []
+
+    for person in people:
+        person_instance = Person(person["name"], person["age"])
+        person_instances.append(person_instance)
+
+    for person in people:
+        person_instance = Person.people[person["name"]]
+        if "wife" in person and person["wife"]:
+            person_instance.wife = Person.people[person["wife"]]
```

Comment:
This line assumes that the 'wife' mentioned in the dictionary is already created and exists in `Person.people`. Ensure that the input list is ordered or structured in a way that all referenced persons are created before they are assigned as a spouse.